src/libraries/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/PemEncoding.cs

```diff
@@ -0,0 +1,518 @@
+        {
+            // Check for the minimum possible encoded length of a PEM structure
+            // and exit early if there is no way the input could contain a well-formed
+            // PEM.
+            if (pemData.Length < PreEBPrefix.Length + Ending.Length * 2 + PostEBPrefix.Length)
+            {
+                fields = default;
+                return false;
+            }
+
+            const int PostebStackBufferSize = 256;
+            Span<char> postebStackBuffer = stackalloc char[PostebStackBufferSize];
+            int areaOffset = 0;
+            int preebIndex;
+            while ((preebIndex = pemData.IndexOfByOffset(PreEBPrefix, areaOffset)) >= 0)
+            {
+                int labelStartIndex = preebIndex + PreEBPrefix.Length;
+
+                // If there are any previous characters, the one prior to the PreEB
+                // must be a white space character.
+                if (preebIndex > 0 && !char.IsWhiteSpace(pemData[preebIndex - 1]))
+                {
+                    areaOffset += labelStartIndex;
+                    continue;
+                }
+
+                int preebEndIndex = pemData.IndexOfByOffset(Ending, labelStartIndex);
+
+                // There is no ending sequence, -----, in the remainder of
+                // the document. Therefore, there can never be a complete PreEB
+                // and we can exit.
+                if (preebEndIndex < 0)
+                {
+                    fields = default;
+                    return false;
+                }
+
+                Range labelRange = labelStartIndex..preebEndIndex;
+                ReadOnlySpan<char> label = pemData[labelRange];
+
+                // There could be a preeb that is valid after this one if it has an invalid
+                // label, so move from there.
+                if (!IsValidLabel(label))
+                {
+                    goto NextAfterLabel;
+                }
+
+                int contentStartIndex = preebEndIndex + Ending.Length;
+                int postebLength = PostEBPrefix.Length + label.Length + Ending.Length;
+
+                Span<char> postebBuffer = postebLength > PostebStackBufferSize
+                    ? new char[postebLength]
+                    : postebStackBuffer;
+                ReadOnlySpan<char> posteb = WritePostEB(label, postebBuffer);
+                int postebStartIndex = pemData.IndexOfByOffset(posteb, contentStartIndex);
+
+                if (postebStartIndex < 0)
+                {
+                    goto NextAfterLabel;
+                }
+
+                int pemEndIndex = postebStartIndex + postebLength;
+
+                // The PostEB must either end at the end of the string, or
+                // have at least one white space character after it.
+                if (pemEndIndex < pemData.Length - 1 &&
+                    !char.IsWhiteSpace(pemData[pemEndIndex]))
```

Comment:
@GrabYourPitchforks Do you have opinions on this "IsWhiteSpace" for determining the acceptability for end-of-payload?

The idea is `-----BEGIN FOO-----base64==----END FOO-----` is legal, `-----BEGIN FOO-----base64==----END FOO-----M` isn't (post-encapsulation-boundary might be part of some other text), but `-----BEGIN FOO-----base64==----END FOO----- M` is.  If the space is U+200A (hair space) is that enough to say we've hit the "end of the word" and thus the end of the post-EB?  Or would you prefer we stick to the RFC's `*W` production and limit it to `{ space, \t, \r, \n, \v, Form-feed }` or even our limited `*W` (`{ space, \t, \r, \n }`)?

Reply:
Generally speaking we should never have a default behavior more _lax_ than the RFC (though we can be more _strict_ by default). This is the same philosophy we also followed with the JSON stack. So I'd suggest limiting this to RFC-blessed whitespace chars.

Reply:
Ah, the _end_ of the payload. Nevermind then. No real opinions one way or another. :)

Reply:
@vcsjones Can you throw in a test that uses \u200A before and after to show that it satisfies the pre-preEB/post-postEB boundary constraints?  Alternatively, change the two IsWhiteSpaces to only allow space-tab-CR-LF and add a test that says we don't support the full set of IsWhiteSpace?

Since both @GrabYourPitchforks and I are iffy, it's your call on permissive or restrictive at the boundary.

Reply:
@bartonjs 
Since the base-64 decoding does not permit `/v` and `/f`, it made sense to me to limit the base-64 check to the more narrow definition of white space. It also makes sense to me to have a single definition of "white space" when decoding, so I changed the pre and post boundaries to match the same definition of white space as the base-64 validation.